File: app/src/main/java/com/iceteaviet/fastfoodfinder/activity/profile/ListDetailActivity.java
package com.iceteaviet.fastfoodfinder.activity.profile;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.iceteaviet.fastfoodfinder.R;
import com.iceteaviet.fastfoodfinder.adapter.StoreListAdapter;
import com.iceteaviet.fastfoodfinder.model.store.StoreDataSource;
import com.iceteaviet.fastfoodfinder.ui.DividerItemDecoration;
import com.iceteaviet.fastfoodfinder.utils.Constant;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import de.hdodenhof.circleimageview.CircleImageView;

/**
 * Created by MyPC on 12/6/2016.
 */
public class ListDetailActivity extends AppCompatActivity {

    @BindView(R.id.rvList)
    RecyclerView rvStoreList;
    @BindView(R.id.iconList)
    CircleImageView cvIconList;
    @BindView(R.id.tvListName)
    TextView tvListName;
    @BindView(R.id.tvNumberPlace)
    TextView tvNumberPlace;
    @BindView(R.id.cvAvatar)
    CircleImageView avatar;
    StoreListAdapter mAdapter;
    LinearLayoutManager layoutManager;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list_detail);
        ButterKnife.bind(this);
        mAdapter = new StoreListAdapter();
        layoutManager = new LinearLayoutManager(getApplicationContext());
        rvStoreList.setAdapter(mAdapter);
        rvStoreList.setLayoutManager(layoutManager);
        RecyclerView.ItemDecoration decoration = new DividerItemDecoration(getApplicationContext(), DividerItemDecoration.VERTICAL_LIST);
        rvStoreList.addItemDecoration(decoration);
        getData();
    }

    void getData() {
        Intent intent = getIntent();

        //list name
        tvListName.setText(intent.getStringExtra(Constant.KEY_NAME));
        //icon of list
        cvIconList.setImageResource(intent.getIntExtra(Constant.KEY_IDICON, 0));

        Glide.with(getApplicationContext())
                .load(intent.getStringExtra(Constant.KEY_URL))
                .into(avatar);
        //id of list
        int id = intent.getIntExtra(Constant.KEY_ID, 0);
        //list id of store
        ArrayList<Integer> list = intent.getIntegerArrayListExtra(Constant.KEY_STORE);

        int numberPlace = list.size();

        //add list store to mAdapter here
        mAdapter.setStores(StoreDataSource.getStoresById(list));
    }
}

File: app/src/main/java/com/iceteaviet/fastfoodfinder/ui/storelist/ListDetailActivity.kt
package com.iceteaviet.fastfoodfinder.ui.storelist

import android.os.Bundle
import android.widget.Toast
import androidx.activity.viewModels
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.repeatOnLifecycle
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import dagger.hilt.android.AndroidEntryPoint
import com.bumptech.glide.Glide
import com.iceteaviet.fastfoodfinder.R
import com.iceteaviet.fastfoodfinder.data.remote.user.model.UserStoreList
import com.iceteaviet.fastfoodfinder.databinding.ActivityListDetailBinding
import com.iceteaviet.fastfoodfinder.ui.base.BaseActivity
import de.hdodenhof.circleimageview.CircleImageView
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

@AndroidEntryPoint
class ListDetailActivity : BaseActivity() {

    private val viewModel: ListDetailViewModel by viewModels()

    private lateinit var binding: ActivityListDetailBinding
    lateinit var rvStoreList: RecyclerView
    lateinit var cvIconList: CircleImageView
    private var mAdapter: StoreListAdapter? = null

    override val layoutId: Int
        get() = R.layout.activity_list_detail

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityListDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)

        if (intent != null) {
            viewModel.handleExtras(
                intent.getParcelableExtra<UserStoreList>(KEY_USER_STORE_LIST), 
                intent.getStringExtra(KEY_USER_PHOTO_URL)
            )
            setupUI()
            setupObservers()
        } else {
            finish()
        }
    }

    private fun setupObservers() {
        lifecycleScope.launch {
            repeatOnLifecycle(Lifecycle.State.STARTED) {
                viewModel.uiState.collectLatest { state ->
                    binding.tvListName.text = state.listName
                    
                    if (state.storeIconResId != 0) {
                        Glide.with(applicationContext)
                            .load(state.storeIconResId)
                            .into(cvIconList)
                    }

                    mAdapter?.setStores(state.stores)

                    when (state.event) {
                        is ListDetailEvent.Idle -> {}
                        is ListDetailEvent.Exit -> {
                            finish()
                            viewModel.markEventConsumed()
                        }
                        is ListDetailEvent.ShowGeneralErrorMessage -> {
                            Toast.makeText(this@ListDetailActivity, R.string.error_general_error_code, Toast.LENGTH_LONG).show()
                            viewModel.markEventConsumed()
                        }
                    }
                }
            }
        }
    }

    override fun onResume() {
        super.onResume()
        viewModel.start()
    }

    private fun setupUI() {
        rvStoreList = binding.rvList
        cvIconList = binding.iconList

        mAdapter = StoreListAdapter()
        val layoutManager = LinearLayoutManager(applicationContext)
        rvStoreList.adapter = mAdapter
        rvStoreList.layoutManager = layoutManager
        val decoration = DividerItemDecoration(applicationContext, DividerItemDecoration.VERTICAL)
        rvStoreList.addItemDecoration(decoration)
    }

    companion object {
        const val KEY_USER_STORE_LIST = "store"
        const val KEY_USER_PHOTO_URL = "url"
    }
}